internal/runner/jest.go

```diff
@@ -0,0 +1,196 @@
+		return RunResult{Status: RunStatusError}, err
+	}
+
+	if exitError := new(exec.ExitError); errors.As(err, &exitError) {
+		report, parseErr := j.ParseReport(j.ResultPath)
+		if parseErr != nil {
+			fmt.Println("Buildkite Test Splitter: Failed to read Jest output, tests will not be retried.")
+			return RunResult{Status: RunStatusError}, err
+		}
+
+		if report.NumFailedTests > 0 {
+			var failedTests []string
+			for _, testResult := range report.TestResults {
+				for _, example := range testResult.AssertionResults {
+					if example.Status == "failed" {
+						failedTests = append(failedTests, example.Name)
+					}
+				}
+			}
+
+			return RunResult{Status: RunStatusFailed, FailedTests: failedTests}, nil
+		}
+	}
+
+	return RunResult{Status: RunStatusError}, err
+}
+
+type JestExample struct {
+	Name   string `json:"fullName"`
+	Status string `json:"status"`
+}
+
+type JestReport struct {
+	NumFailedTests int
+	TestResults    []struct {
+		AssertionResults []JestExample
+	}
+}
+
+func (j Jest) ParseReport(path string) (JestReport, error) {
+	var report JestReport
+	data, err := os.ReadFile(path)
+	if err != nil {
+		return JestReport{}, fmt.Errorf("failed to read Jest output: %v", err)
+	}
+
+	if err := json.Unmarshal(data, &report); err != nil {
+		return JestReport{}, fmt.Errorf("failed to parse Jest output: %s", err)
+	}
+
+	return report, nil
+}
+
+func (j Jest) commandNameAndArgs(cmd string, testCases []string) (string, []string, error) {
+	words, err := shellquote.Split(cmd)
+	if err != nil {
+		return "", []string{}, err
+	}
+	idx := slices.Index(words, "{{testExamples}}")
+	if idx < 0 {
+		words = append(words, testCases...)
+	} else {
+		words = slices.Replace(words, idx, idx+1, testCases...)
+	}
+
+	outputIdx := slices.Index(words, "{{outputFile}}")
+	if outputIdx < 0 {
+		err := fmt.Errorf("couldn't find '{{outputFile}}' sentinel in command, exiting.")
+		return "", []string{}, err
+	}
+	slices.Replace(words, outputIdx, outputIdx+1, j.ResultPath)
+
+	return words[0], words[1:], nil
+}
+
+func (j Jest) retryCommandNameAndArgs(cmd string, testCases []string) (string, []string, error) {
+	words, err := shellquote.Split(cmd)
+	if err != nil {
+		return "", []string{}, err
+	}
+
+	idx := slices.Index(words, "{{testNamePattern}}")
+	if idx < 0 {
+		err := fmt.Errorf("couldn't find '{{testNamePattern}}' sentinel in retry command")
+		return "", []string{}, err
+	}
+
+	testNamePattern := fmt.Sprintf("(%s)", strings.Join(testCases, "|"))
+
+	words = slices.Replace(words, idx, idx+1, testNamePattern)
+
+	outputIdx := slices.Index(words, "{{outputFile}}")
+	if outputIdx < 0 {
+		err := fmt.Errorf("couldn't find '{{outputFile}}' sentinel in retry command, exiting.")
+		return "", []string{}, err
+	}
+	slices.Replace(words, outputIdx, outputIdx+1, j.ResultPath)
+
+	return words[0], words[1:], err
+}
+
+func (j Jest) GetExamples(files []string) ([]plan.TestCase, error) {
+	return nil, nil
```

Comment:
Maybe return an error for now?
```suggestion
	return nil, fmt.Error("not supported in Jest")
```